### PR TITLE
fix #285307: JACK Audio on Musescore 3.x not working

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
syncing our jackweakapi.cpp with [https://github.com/jackaudio/jack2](https://github.com/jackaudio/jack2)'s [JackWeakAPI.c](https://github.com/jackaudio/jack2/blob/develop/common/JackWeakAPI.c) (commenting out new(?) APIs we most probably don't use anyhow).
This minimizes the diffs between those 2 files and should bring MuseScore 3 64bit back to play along with Jack.

### CODE REVIEW

**Comment by MarcSabatella** [comment]
I couldn't get the AppVeyor build of this to work with JACK, but I'm hardly a JACK expert.  My own 32-bit builds *do* work (from master, I mean, not including these changes) so I know it's at least *possible* on my system.  I have the 64-bit JACK installed, which supposedly (?) can now handle both 32 & 64 bit applications.

File: mscore/jackweakapi.cpp [hunk 1]
```diff
@@ -74,16 +73,27 @@ static int tryload_libjack()
...
+            qDebug("dlopen error : %s ", dlerror());
+        }
     #elif defined(WIN32)
         // Force char implementation of library instead of possibly wchar_t implementation to be called.
-        libjack_handle = LoadLibraryA("libjack.dll");
+        #ifdef _WIN64
+            libjack_handle = LoadLibraryA("libjack64.dll");
+        #else
+            libjack_handle = LoadLibraryA("libjack.dll");
+        #endif
```
Comment (dmitrio95) [inline mscore/jackweakapi.cpp]: So basically this is the change that resolves the issue on Windows? Other changes seem to be mostly about functions which are commented out anyway.

Comment (Jojo-Schmitz) [reply mscore/jackweakapi.cpp]: Yes

Comment (dmitrio95) [reply mscore/jackweakapi.cpp]: Well, OK, that would be curious if the issue has been just in the library name. Let's merge it then and see if it helps for others.

**Comment by rgeddes** [comment]
I'm using MuseScore 3.62 installed from snap on Ubuntu 20.04, and when I try to select jack audio in the I/O section the musescore crashes.  On the jack side I see no errors or attempts to connect from musescore.

**Comment by Jojo-Schmitz** [comment]
The snap version is not supported here or on musescore.org, but only on snap website

**Pull request merged**